src/process_utility.c

```diff
@@ -309,6 +311,137 @@ process_reindex(Node *parsetree)
 }
 
 
+static void
+process_altertable_change_owner(Cache *hcache, AlterTableCmd *cmd, Oid relid)
+{
+	Hypertable *ht;
+	RoleSpec   *role;
+
+	ht = hypertable_cache_get_entry(hcache, relid);
+	if (NULL == ht)
+		return;
+
+	Assert(IsA(cmd->newowner, RoleSpec));
+	role = (RoleSpec *) cmd->newowner;
+	OidFunctionCall2(catalog_get_internal_function_id(catalog_get(), DDL_CHANGE_OWNER),
+				   ObjectIdGetDatum(relid), CStringGetDatum(role->rolename));
+}
+
+static void
+process_altertable_add_constraint(Cache *hcache, AlterTableCmd *cmd, Oid relid)
+{
+	Hypertable *ht;
+	char	   *constraint_name = NULL;
+
+	ht = hypertable_cache_get_entry(hcache, relid);
+	if (NULL == ht)
+		return;
+
+	if (cmd->name != NULL)
+	{
+		constraint_name = cmd->name;
+	}
+	else
+	{
+		if (cmd->subtype == AT_AddIndex)
+		{
+			Node	   *def = cmd->def;
+
+			if (IsA(def, IndexStmt))
+			{
+				IndexStmt  *stmt = (IndexStmt *) def;
+
+				Assert(stmt->isconstraint);
+				constraint_name = stmt->idxname;
+			}
+		}
+		if (cmd->subtype == AT_AddConstraint
+			|| cmd->subtype == AT_AddConstraintRecurse)
+		{
+			Node	   *def = cmd->def;
+
+			if (IsA(def, Constraint))
+			{
+				Constraint *stmt = (Constraint *) def;
+
+				constraint_name = stmt->conname;
+			}
+
+		}
+	}
+
+	Assert(constraint_name != NULL);
+	OidFunctionCall2(catalog_get_internal_function_id(catalog_get(), DDL_ADD_CONSTRAINT),
```

Comment:
Yeah that's  what I fundamentally don't get --- why are upcalls a last resort. The resulting code is much more readable and compact with upcalls then without.